### PR TITLE
Fix lazy initialization in RabbitMQ

### PR DESCRIPTION
**Pull request by Avogar**
### Changelog category (leave one):
- Bug Fix (user-visible misbehavior in an official stable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fix lazy initialization in RabbitMQ that could lead to logical error and not initialized state.

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

File: src/Storages/IMessageProducer.cpp [hunk 1]
```diff
@@ -12,7 +12,16 @@ void AsynchronousMessageProducer::start(const ContextPtr & context)
...
-    initialize();
+    try
+    {
+        initialize();
+    }
+    catch (...)
+    {
+        finished = true;
+        throw;
+    }
```
Comment (Avogar) [inline src/Storages/IMessageProducer.cpp:23]: Previously if we tried to insert into RabbitMQ/NATS that is not initialized, it led to deadlock in `AsynchronousMessageProducer::finish` on `schedule.wait()`. This fixes it

**Comment by robot-clickhouse-ci-2** [comment]
*This is an automated comment for commit 3a5ba56c36775709d8cfa4b2d6deb081b69dd2a5 with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/59352/3a5ba56c36775709d8cfa4b2d6deb081b69dd2a5/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/59352/3a5ba56c36775709d8cfa4b2d6deb081b69dd2a5/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

**Review by kssenii — APPROVED** [review]

**Review by alexey-milovidov — APPROVED** [review]

**Comment by Algunenano** [comment]
TSAN is not happy -> https://s3.amazonaws.com/clickhouse-test-reports/59388/6e82dc81fcf6daa0180ec09f82a1e367788864f6/integration_tests__tsan__[4_6]//home/ubuntu/actions-runner/_work/_temp/test/output_dir/integration_run_parallel3_0.log

The fact that we had partially disabled TSAN runs probably doesn't help detecting the problem. @Avogar Can you have a look, please?

**Comment by Avogar** [comment]
@Algunenano https://github.com/ClickHouse/ClickHouse/issues/59415#issuecomment-1921895577
We should update `libuv` contrib

**Pull request merged**